### PR TITLE
Fixed #1548 : Added Documentation regarding Contributions Best Practices

### LINKED ISSUE #1548: No Documentation regarding Contributions Best Practices
It would be great If Documentation regarding Contributions Best Practices so it would be easy for the newcomers.

### PR DESCRIPTION
**Pull request by naveentherocker**
Fixes #1548 
Changes:
Added Documentation regarding Contributions Best Practices in CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,6 +14,23 @@ Please note we have a code of conduct, please follow it in all your interactions
...
+* When you're submitting a PR for a UI-related issue, it would be really awesome if you add a screenshot of your change or a link to a deployment where it can be tested out along with your PR. It makes it very easy for the reviewers and you'll also get reviews quicker.
+
+**Feature Requests and Bug Reports**
+* When you file a feature request or when you are submitting a bug report to the [issue tracker](https://github.com/fossasia/phimpme-android/issues), make sure you add steps to reproduce it. Especially if that bug is some weird/rare one.
+
+**Join the development**
+* Before you join development, please set up the project on your local machine, run it and go through the application completely. Press on any button you can find and see where it leads to. Explore. (Don't worry ... Nothing will happen to the app or to you due to the exploring :wink: Only thing that will happen is, you'll be more familiar with what is where and might even get some cool ideas on how to improve various aspects of the app.)
```
Comment (anantprsd5) [inline CONTRIBUTING.md]: Remove "Press on any ... aspects of the app." Please don't make use of any winks and all in this file. Try to keep it formal. Thanks

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,6 +14,23 @@ Please note we have a code of conduct, please follow it in all your interactions
...
 
+## Contributions Best Practices
+
+**Commits**
+* Write clear meaningful git commit messages (Do read http://chris.beams.io/posts/git-commit/)
+* Make sure your PR's description contains GitHub's special keyword references that automatically close the related issue when the PR is merged. (More info at https://github.com/blog/1506-closing-issues-via-pull-requests )
+* When you make very very minor changes to a PR of yours (like for example fixing a failing travis build or some small style corrections or minor changes requested by reviewers) make sure you squash your commits afterwards so that you don't have an absurd number of commits for a very small fix. (Learn how to squash at https://davidwalsh.name/squash-commits-git )
```
Comment (anantprsd5) [inline CONTRIBUTING.md]: Very minor.

File: CONTRIBUTING.md (hunk 1 above)
Comment (naveentherocker) [reply CONTRIBUTING.md]: @anantprsd5 Most of the FOSSASIA's Repos are using the Same CONTRIBUTING.md

File: CONTRIBUTING.md (hunk 2 above)
Comment (naveentherocker) [reply CONTRIBUTING.md]: I think these are Important for a Beginner to Contribute.

**Comment by naveentherocker** [comment]
@anantprsd5 Please reply

**Comment by anantprsd5** [comment]
@naveentherocker I don't know about all the projects but making use of informal languages in an official documentation doesn't seem to be a good practice. Please make the suggested changes. Thanks.

**Comment by anantprsd5** [comment]
@naveentherocker Also send different PR for the codecov issue.

**Comment by naveentherocker** [comment]
Ok sure.

**Pull request merged**